lisa/tools/sar.py

```diff
@@ -0,0 +1,114 @@
+        #     EDEV    Network interfaces (errors)
+        #     NFS     NFS client
+        #     NFSD    NFS server
+        #     SOCK    Sockets (v4)
+        #     IP      IP traffic      (v4)
+        #     EIP     IP traffic      (v4) (errors)
+        #     ICMP    ICMP traffic    (v4)
+        #     EICMP   ICMP traffic    (v4) (errors)
+        #     TCP     TCP traffic     (v4)
+        #     ETCP    TCP traffic     (v4) (errors)
+        #     UDP     UDP traffic     (v4)
+        #     SOCK6   Sockets (v6)
+        #     IP6     IP traffic      (v6)
+        #     EIP6    IP traffic      (v6) (errors)
+        #     ICMP6   ICMP traffic    (v6)
+        #     EICMP6  ICMP traffic    (v6) (errors)
+        #     UDP6    UDP traffic     (v6)
+        #     FC      Fibre channel HBAs
+        #     SOFT    Software-based network processing
+        cmd = f"{self.command} -n {key_word} {interval} {count}"
+        process = self.node.execute_async(cmd)
+        return process
+
+    def run_with(
+        self, key_word: str = "DEV", interval: int = 1, count: int = 300
+    ) -> NetworkPPSPerformanceMessage:
+        process = self.run_with_parameter_async(key_word, interval, count)
+        timeout = count + 10
+        process_result = process.wait_result(
+            timeout,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="fail to run sar command",
+        )
+        # IFACE: Name of the network interface for which statistics are reported.
+        # rxpck/s: packet receiving rate (unit: packets/second)
+        # txpck/s: packet transmitting rate (unit: packets/second)
+        # rxkB/s: data receiving rate (unit: Kbytes/second)
+        # txkB/s: data transmitting rate (unit: Kbytes/second)
+        # rxcmp/s: compressed packets receiving rate (unit: Kbytes/second)
+        # txcmp/s: compressed packets transmitting rate (unit: Kbytes/second)
+        # rxmcst/s: multicast packets receiving rate (unit: Kbytes/second)
+        nic_name = self.node.nics.default_nic
+        sar_result_pattern = re.compile(
+            rf"([\w\W]*?){nic_name}\s+(?P<rxpck>\d+.\d+)\s+(?P<txpck>\d+.\d+)"
+            r"\s+(?P<rxkb>\d+.\d+)\s+(?P<rxcmp>\d+.\d+)\s+(?P<txcmp>\d+.\d+)"
+            r"\s+(?P<rxmcst>\d+.\d+)\s+(?P<ifutil>\d+.\d+)",
+            re.M,
+        )
+        raw_list = re.finditer(self.sar_results_pattern, process_result.stdout)
+        rx_pps: List[Decimal] = []
+        tx_pps: List[Decimal] = []
+        tx_rx_pps: List[Decimal] = []
+        for sar_result in raw_list:
+            temp = sar_result_pattern.match(sar_result.group())
+            assert temp, f"not find matched sar result for nic {nic_name}"
+            rx_pps.append(Decimal(temp.group("rxpck")))
+            tx_pps.append(Decimal(temp.group("txpck")))
+            tx_rx_pps.append(
+                Decimal(temp.group("rxpck")) + Decimal(temp.group("txpck"))
+            )
+        message = NetworkPPSPerformanceMessage()
```

Comment:
If we construct message in this method, this method can deal with network message only. So, either change this method to `get_network_statictics`, or return a dict for general purpose. I suggest trying first way, it doesn't need to parse results of other types.

Reply:
updated